# Verzeo Project: EDA (TMDb Movie Data)

## Questions:
### 1) Which are the movies with most and least earned revenue?
### 2) What is the average runtime of movies in the year 2006?
### 3) Name any 3 production companies which have invested money in worse revenue movies?
### 4) Which are the movies with the third lowest and third highest budget?


In [1]:
#Importing the essential libraries
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('tmdb-movies.csv')   
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10866 entries, 0 to 10865
Data columns (total 21 columns):
id                      10866 non-null int64
imdb_id                 10856 non-null object
popularity              10866 non-null float64
budget                  10866 non-null int64
revenue                 10866 non-null int64
original_title          10866 non-null object
cast                    10790 non-null object
homepage                2936 non-null object
director                10822 non-null object
tagline                 8042 non-null object
keywords                9373 non-null object
overview                10862 non-null object
runtime                 10866 non-null int64
genres                  10843 non-null object
production_companies    9836 non-null object
release_date            10866 non-null object
vote_count              10866 non-null int64
vote_average            10866 non-null float64
release_year            10866 non-null int64
budget_adj              1

In [3]:
df

id    imdb_id  popularity     budget     revenue  \
0      135397  tt0369610   32.985763  150000000  1513528810   
1       76341  tt1392190   28.419936  150000000   378436354   
2      262500  tt2908446   13.112507  110000000   295238201   
3      140607  tt2488496   11.173104  200000000  2068178225   
4      168259  tt2820852    9.335014  190000000  1506249360   
5      281957  tt1663202    9.110700  135000000   532950503   
6       87101  tt1340138    8.654359  155000000   440603537   
7      286217  tt3659388    7.667400  108000000   595380321   
8      211672  tt2293640    7.404165   74000000  1156730962   
9      150540  tt2096673    6.326804  175000000   853708609   
10     206647  tt2379713    6.200282  245000000   880674609   
11      76757  tt1617661    6.189369  176000003   183987723   
12     264660  tt0470752    6.118847   15000000    36869414   
13     257344  tt2120120    5.984995   88000000   243637091   
14      99861  tt2395427    5.944927  280000000  1405035767   
15     273248  tt3460252    5.898400   44000000   155760117   
16     260346  tt2446042    5.749758   48000000   325771424   
17     102899  tt0478970    5.573184  130000000   518602163   
18     150689  tt1661199    5.556818   95000000   542351353   
19     131634  tt1951266    5.476958  160000000   650523427   
20     158852  tt1964418    5.462138  190000000   209035668   
21     307081  tt1798684    5.337064   30000000    91709827   
22     254128  tt2126355    4.907832  110000000   470490832   
23     216015  tt2322441    4.710402   40000000   569651467   
24     318846  tt1596363    4.648046   28000000   133346506   
25     177677  tt2381249    4.566713  150000000   682330139   
26     214756  tt2637276    4.564549   68000000   215863606   
27     207703  tt2802144    4.503789   81000000   403802136   
28     314365  tt1895587    4.062293   20000000    88346473   
29     294254  tt4046784    3.968891   61000000   311256926   
...       ...        ...         ...        ...         ...   
10836   38720  tt0061170    0.239435          0           0   
10837   19728  tt0060177    0.291704          0           0   
10838   22383  tt0060862    0.151845          0           0   
10839   13353  tt0060550    0.276133          0           0   
10840   34388  tt0060437    0.102530          0           0   
10841   42701  tt0062262    0.264925      75000           0   
10842   36540  tt0061199    0.253437          0           0   
10843   29710  tt0060588    0.252399          0           0   
10844   23728  tt0059557    0.236098          0           0   
10845    5065  tt0059014    0.230873          0           0   
10846   17102  tt0059127    0.212716          0           0   
10847   28763  tt0060548    0.034555          0           0   
10848    2161  tt0060397    0.207257    5115000    12000000   
10849   28270  tt0060445    0.206537          0           0   
10850   26268  tt0060490    0.202473          0           0   
10851   15347  tt0060182    0.342791          0           0   
10852   37301  tt0060165    0.227220          0           0   
10853   15598  tt0060086    0.163592          0           0   
10854   31602  tt0060232    0.146402          0           0   
10855   13343  tt0059221    0.141026     700000           0   
10856   20277  tt0061135    0.140934          0           0   
10857    5921  tt0060748    0.131378          0           0   
10858   31918  tt0060921    0.317824          0           0   
10859   20620  tt0060955    0.089072          0           0   
10860    5060  tt0060214    0.087034          0           0   
10861      21  tt0060371    0.080598          0           0   
10862   20379  tt0060472    0.065543          0           0   
10863   39768  tt0060161    0.065141          0           0   
10864   21449  tt0061177    0.064317          0           0   
10865   22293  tt0060666    0.035919      19000           0   

                                         original_title  \
0                                        Jura

In [4]:
df.describe()

id    popularity        budget       revenue       runtime  \
count   10866.000000  10866.000000  1.086600e+04  1.086600e+04  10866.000000   
mean    66064.177434      0.646441  1.462570e+07  3.982332e+07    102.070863   
std     92130.136561      1.000185  3.091321e+07  1.170035e+08     31.381405   
min         5.000000      0.000065  0.000000e+00  0.000000e+00      0.000000   
25%     10596.250000      0.207583  0.000000e+00  0.000000e+00     90.000000   
50%     20669.000000      0.383856  0.000000e+00  0.000000e+00     99.000000   
75%     75610.000000      0.713817  1.500000e+07  2.400000e+07    111.000000   
max    417859.000000     32.985763  4.250000e+08  2.781506e+09    900.000000   

         vote_count  vote_average  release_year    budget_adj   revenue_adj  
count  10866.000000  10866.000000  10866.000000  1.086600e+04  1.086600e+04  
mean     217.389748      5.974922   2001.322658  1.755104e+07  5.136436e+07  
std      575.619058      0.935142     12.812941  3.430616e+07  1.446325e+08  
min       10.000000      1.500000   1960.000000  0.000000e+00  0.000000e+00  
25%       17.000000      5.400000   1995.000000  0.000000e+00  0.000000e+00  
50%       38.000000      6.000000   2006.000000  0.000000e+00  0.000000e+00  
75%      145.750000      6.600000   2011.000000  2.085325e+07  3.369710e+07  
max     9767.000000      9.200000   2015.000000  4.250000e+08  2.827124e+09

In [5]:
df.isna().sum()

id                         0
imdb_id                   10
popularity                 0
budget                     0
revenue                    0
original_title             0
cast                      76
homepage                7930
director                  44
tagline                 2824
keywords                1493
overview                   4
runtime                    0
genres                    23
production_companies    1030
release_date               0
vote_count                 0
vote_average               0
release_year               0
budget_adj                 0
revenue_adj                0
dtype: int64

# Data Cleaning (Removing The Unused Information From The Dataset)

In [6]:
refined_df = df.drop(['budget_adj','revenue_adj','overview','imdb_id','homepage','tagline','popularity','director','keywords','release_date','vote_count','vote_average','cast','genres'],axis =1)

In [7]:
refined_df

id     budget     revenue  \
0      135397  150000000  1513528810   
1       76341  150000000   378436354   
2      262500  110000000   295238201   
3      140607  200000000  2068178225   
4      168259  190000000  1506249360   
5      281957  135000000   532950503   
6       87101  155000000   440603537   
7      286217  108000000   595380321   
8      211672   74000000  1156730962   
9      150540  175000000   853708609   
10     206647  245000000   880674609   
11      76757  176000003   183987723   
12     264660   15000000    36869414   
13     257344   88000000   243637091   
14      99861  280000000  1405035767   
15     273248   44000000   155760117   
16     260346   48000000   325771424   
17     102899  130000000   518602163   
18     150689   95000000   542351353   
19     131634  160000000   650523427   
20     158852  190000000   209035668   
21     307081   30000000    91709827   
22     254128  110000000   470490832   
23     216015   40000000   569651467   
24     318846   28000000   133346506   
25     177677  150000000   682330139   
26     214756   68000000   215863606   
27     207703   81000000   403802136   
28     314365   20000000    88346473   
29     294254   61000000   311256926   
...       ...        ...         ...   
10836   38720          0           0   
10837   19728          0           0   
10838   22383          0           0   
10839   13353          0           0   
10840   34388          0           0   
10841   42701      75000           0   
10842   36540          0           0   
10843   29710          0           0   
10844   23728          0           0   
10845    5065          0           0   
10846   17102          0           0   
10847   28763          0           0   
10848    2161    5115000    12000000   
10849   28270          0           0   
10850   26268          0           0   
10851   15347          0           0   
10852   37301          0           0   
10853   15598          0           0   
10854   31602          0           0   
10855   13343     700000           0   
10856   20277          0           0   
10857    5921          0           0   
10858   31918          0           0   
10859   20620          0           0   
10860    5060          0           0   
10861      21          0           0   
10862   20379          0           0   
10863   39768          0           0   
10864   21449          0           0   
10865   22293      19000           0   

                                         original_title  runtime  \
0                                        Jurassic World      124   
1                                    Mad Max: Fury Road      120   
2                                             Insurgent      119   
3                          Star Wars: The Force Awakens      136   
4                                             Furious 7      137   
5                                          The Revenant      156   
6                                    Terminator Genisys      125   
7                                           The Martian      141   
8                                               Minions       91   
9                                            Inside Out       94   
10                                              Spectre      148   
11                                    Jupiter Ascending      124   
12                                           Ex Machina      108   
13                                               Pixels      105   
14                              Avengers: Age of Ultron      141   
15                                    The Hateful Eight      167   
16                                              Taken 3      109   
17                                              Ant-Man      115   
18                                           Cinderella      112   
19                The Hunger Games: Mockingjay - Part 2      136   
20                                         Tomorrowland      130   
21                                

In [8]:
#Removing null values
refined_df.dropna(inplace=True)
refined_df

id     budget     revenue  \
0      135397  150000000  1513528810   
1       76341  150000000   378436354   
2      262500  110000000   295238201   
3      140607  200000000  2068178225   
4      168259  190000000  1506249360   
5      281957  135000000   532950503   
6       87101  155000000   440603537   
7      286217  108000000   595380321   
8      211672   74000000  1156730962   
9      150540  175000000   853708609   
10     206647  245000000   880674609   
11      76757  176000003   183987723   
12     264660   15000000    36869414   
13     257344   88000000   243637091   
14      99861  280000000  1405035767   
15     273248   44000000   155760117   
16     260346   48000000   325771424   
17     102899  130000000   518602163   
18     150689   95000000   542351353   
19     131634  160000000   650523427   
20     158852  190000000   209035668   
21     307081   30000000    91709827   
22     254128  110000000   470490832   
23     216015   40000000   569651467   
24     318846   28000000   133346506   
25     177677  150000000   682330139   
26     214756   68000000   215863606   
27     207703   81000000   403802136   
28     314365   20000000    88346473   
29     294254   61000000   311256926   
...       ...        ...         ...   
10834   12639          0           0   
10835    5923   12000000    20000000   
10836   38720          0           0   
10837   19728          0           0   
10838   22383          0           0   
10839   13353          0           0   
10840   34388          0           0   
10841   42701      75000           0   
10843   29710          0           0   
10844   23728          0           0   
10845    5065          0           0   
10846   17102          0           0   
10847   28763          0           0   
10848    2161    5115000    12000000   
10849   28270          0           0   
10850   26268          0           0   
10851   15347          0           0   
10852   37301          0           0   
10854   31602          0           0   
10855   13343     700000           0   
10856   20277          0           0   
10857    5921          0           0   
10858   31918          0           0   
10859   20620          0           0   
10860    5060          0           0   
10861      21          0           0   
10862   20379          0           0   
10863   39768          0           0   
10864   21449          0           0   
10865   22293      19000           0   

                                         original_title  runtime  \
0                                        Jurassic World      124   
1                                    Mad Max: Fury Road      120   
2                                             Insurgent      119   
3                          Star Wars: The Force Awakens      136   
4                                             Furious 7      137   
5                                          The Revenant      156   
6                                    Terminator Genisys      125   
7                                           The Martian      141   
8                                               Minions       91   
9                                            Inside Out       94   
10                                              Spectre      148   
11                                    Jupiter Ascending      124   
12                                           Ex Machina      108   
13                                               Pixels      105   
14                              Avengers: Age of Ultron      141   
15                                    The Hateful Eight      167   
16                                              Taken 3      109   
17                                              Ant-Man      115   
18                                           Cinderella      112   
19                The Hunger Games: Mockingjay - Part 2      136   
20                                         Tomorrowland      130   
21                                

In [9]:
#A seperate dataframe for questions related to budget
refined_df_budget = refined_df[refined_df['budget']>0]
print(refined_df_budget.shape)

(5034, 7)


In [10]:
#A seperate dataframe for questions related to revenue
refined_df_revenue = refined_df[refined_df['revenue']>0]
print(refined_df_revenue.shape)

(4754, 7)


In [11]:
#A seperate dataframe for questions related to production company
prod_company = refined_df_revenue.drop(['budget','runtime','release_year'],axis =1)
prod_company

id     revenue                         original_title  \
0      135397  1513528810                         Jurassic World   
1       76341   378436354                     Mad Max: Fury Road   
2      262500   295238201                              Insurgent   
3      140607  2068178225           Star Wars: The Force Awakens   
4      168259  1506249360                              Furious 7   
5      281957   532950503                           The Revenant   
6       87101   440603537                     Terminator Genisys   
7      286217   595380321                            The Martian   
8      211672  1156730962                                Minions   
9      150540   853708609                             Inside Out   
10     206647   880674609                                Spectre   
11      76757   183987723                      Jupiter Ascending   
12     264660    36869414                             Ex Machina   
13     257344   243637091                                 Pixels   
14      99861  1405035767                Avengers: Age of Ultron   
15     273248   155760117                      The Hateful Eight   
16     260346   325771424                                Taken 3   
17     102899   518602163                                Ant-Man   
18     150689   542351353                             Cinderella   
19     131634   650523427  The Hunger Games: Mockingjay - Part 2   
20     158852   209035668                           Tomorrowland   
21     307081    91709827                               Southpaw   
22     254128   470490832                            San Andreas   
23     216015   569651467                   Fifty Shades of Grey   
24     318846   133346506                          The Big Short   
25     177677   682330139     Mission: Impossible - Rogue Nation   
26     214756   215863606                                  Ted 2   
27     207703   403802136           Kingsman: The Secret Service   
28     314365    88346473                              Spotlight   
29     294254   311256926         Maze Runner: The Scorch Trials   
...       ...         ...                                    ...   
10728     576      638641                         The Wild Bunch   
10740   17529    14250000                              True Grit   
10755     621   181813770                                 Grease   
10756     579   187884007                                 Jaws 2   
10757     923    55000000                       Dawn of the Dead   
10758    1924   300218018                               Superman   
10759     948    70000000                              Halloween   
10760    8469   141000000                           Animal House   
10761    6081    49579269            Revenge of the Pink Panther   
10762   11778    50000000                        The Deer Hunter   
10764   11455    41590893                            Up in Smoke   
10766   23916     7185518                         Paradise Alley   
10770   11327    35000000                       Midnight Express   
10771     123    30471420                  The Lord of the Rings   
10775    4192    14560084                      Death on the Nile   
10777   28932          11                               F.I.S.T.   
10778   17339     7230000                 Force 10 from Navarone   
10779    9584    22765081                                 Convoy   
10780   11850    24046533         Invasion of the Body Snatchers   
10788   24961    21049053                                The Wiz   
10791   10766    26518355                        Damien: Omen II   
10793   11837     3713768                         Watership Down   
10812   31948     5438927                         Gray Lady Down   
10817   13963      321952                         The Last Waltz   
10819   16214    78000000                                 Hooper   
10822     396    33736689        Who's Afraid of Virginia Woolf?   
10828    5780    13000000                           Torn Curtain   

Since now we have the columns, rows and format of the dataset in right way, its time to investigate the data for the questions asked.

# Exploratory Data Analysis

# Question 1 : Which are the movies with most and least earned revenue?

In [12]:
def find_minmax(x):
    #use the function 'idmin' to find the index of lowest revenue movie.
    min_index = refined_df_revenue[x].idxmin()
    #use the function 'idmax' to find the index of Highest revenue movie.
    high_index = refined_df_revenue[x].idxmax()
    high = pd.DataFrame(refined_df_revenue.loc[high_index,:])
    low = pd.DataFrame(refined_df_revenue.loc[min_index,:])
    #print the movie with high and low revenue
    print("Movie Which Has Highest "+ x + " : ",refined_df_revenue['original_title'][high_index])
    print("Movie Which Has Lowest "+ x + "  : ",refined_df_revenue['original_title'][min_index])
    return pd.concat([high,low],axis = 1)
find_minmax('revenue')

Movie Which Has Highest revenue :  Avatar
Movie Which Has Lowest revenue  :  Shattered Glass


1386  \
id                                                                19995   
budget                                                        237000000   
revenue                                                      2781505847   
original_title                                                   Avatar   
runtime                                                             162   
production_companies  Ingenious Film Partners|Twentieth Century Fox ...   
release_year                                                       2009   

                                                                   5067  
id                                                                13537  
budget                                                          6000000  
revenue                                                               2  
original_title                                          Shattered Glass  
runtime                                                              94  
production_companies  Lions Gate Films|Cruise/Wagner Productions|Bau...  
release_year                                                       2003

# Question 2 : What is the average runtime of movies in the year 2006?

In [13]:
refined_df.groupby('release_year')
refined_df_filtered = refined_df.query('release_year == 2006')
mean = refined_df_filtered.groupby('release_year').mean()['runtime']
mean

release_year
2006    101.195467
Name: runtime, dtype: float64

# Question 3 : Name any 3 production companies which have invested money in worse revenue movies?

In [14]:
tester = pd.DataFrame(prod_company.groupby("production_companies").mean()['revenue'])

In [15]:
tester

revenue
production_companies                                            
10th Hole Productions|Benaroya Pictures|Four of...  1.238200e+05
120dB Films|Scion Films|Voodoo Production Servi...  1.378000e+03
1492 Pictures|Dune Entertainment|Fox 2000 Pictu...  1.745788e+08
1492 Pictures|Warner Bros.|Heyday Films             9.764756e+08
1492 Pictures|Warner Bros.|Heyday Films|MIRACLE...  8.766885e+08
1492 Pictures|Warner Bros.|Heyday Films|P of A ...  7.898046e+08
1818|Lone Wolf McQuade Associates|Topkick Produ...  1.223263e+07
20th Century Fox                                    7.108200e+07
20th Century Fox|1492 Pictures                      5.409517e+06
20th Century Fox|American Entertainment Partner...  2.520546e+07
20th Century Fox|Baltimore Pictures                 2.327893e+07
20th Century Fox|Chernin Entertainment|Walden M...  1.197722e+08
20th Century Fox|Davis Entertainment                2.243328e+07
20th Century Fox|Double Feature Films|Appian Wa...  6.261665e+07
20th Century Fox|Egg Pictures                       1.066838e+08
20th Century Fox|Figment Films                      1.440569e+08
20th Century Fox|Gladden Entertainment              4.272120e+07
20th Century Fox|Gran Via Productions|Walden Media  5.310554e+06
20th Century Fox|Gruskoff/Venture Films             2.269769e+07
20th Century Fox|Largo Entertainment|JVC Entert...  5.713872e+07
20th Century Fox|Morgan Creek Productions           2.938196e+07
20th Century Fox|Penta Pictures                     7.549350e+05
20th Century Fox|Regency Enterprises                4.925430e+07
20th Century Fox|Regency Enterprises|Guy Walks ...  5.065008e+07
20th Century Fox|Regency Enterprises|New Regenc...  4.398989e+06
20th Century Fox|Saban Entertainment                1.797974e+07
20th Century Fox|Sandollar Productions              2.162100e+07
20th Century Fox|Turner Pictures                    1.367069e+07
20th Century Fox|Village Roadshow Pictures|NPV ...  1.000201e+08
20th Century Fox|Weed Road Pictures|Dutch Oven|...  4.782075e+08
...                                                          ...
Working Title Films|Translux                        8.710045e+07
Worldview Entertainment                             6.427467e+04
Worldview Entertainment|Arcade Pictures             9.221000e+03
Worldview Entertainment|Dragonfly Entertainment...  6.833445e+06
Worldview Entertainment|Muskat Filmed Propertie...  2.365467e+06
Worldview Entertainment|New Regency Pictures|TS...  1.032151e+08
XYZ Films                                           4.997000e+04
Yari Film Group                                     4.534909e+06
Yari Film Group|Furst Films                         3.822670e+05
Yash Raj Films                                      4.639643e+07
Zazen Produções                                   2.172010e+05
Zenith Entertainment|Initial Pictures|U.K. Prod...  2.826523e+06
Zentropa Entertainments                             2.227167e+06
Zentropa Entertainments|BBC Films|Scottish Scre...  9.080000e+05
Zentropa Entertainments|Danmarks Radio (DR)|Art...  6.749180e+05
Zentropa Entertainments|Det Danske Filminstitut...  7.918670e+05
Zentropa Entertainments|Les Films du Losange|Co...  1.028590e+07
Zero Gravity Management|Farraj Factory|Kinetic ...  2.710000e+03
Zininsa Film Production|Intellectual Properties...  4.040588e+06
Zoetrope Studios                                    8.347151e+07
Zoetrope Studios|Golan-Globus Productions           3.221568e+06
Zucker Brothers Productions                         7.027290e+06
Zweites Deutsches Fernsehen (ZDF)|Les Films Ari...  7.720000e+07
abramoff Production|Scorpion Film Production|Ab...  4.192440e+06
area japan                                          3.170000e+05
chateau productions                                 1.100000e+01
creative differences|History films                  6.467348e+06
erbp                                                4.440980e+05
iDeal Partners Film Fund                            5.512500e+04
icon                       

In [16]:
#Assumption: Movies w least revenue made are taken as context in order to find the production companies.
worst_reader = prod_company.nsmallest(3, ['revenue'])
Top3_worst_prod_comp = worst_reader['production_companies']
Top3_worst_prod_comp.to_frame()

production_companies
5067  Lions Gate Films|Cruise/Wagner Productions|Bau...
8142  Gramercy Pictures|Alphaville Films|View Askew ...
1380                                 Universal Pictures

In [17]:
print("Top 3 worst production companies : ",worst_reader['production_companies'])

Top 3 worst production companies :  5067    Lions Gate Films|Cruise/Wagner Productions|Bau...
8142    Gramercy Pictures|Alphaville Films|View Askew ...
1380                                   Universal Pictures
Name: production_companies, dtype: object


# Question 4 : Which are the movies with the third lowest and third highest budget?

In [18]:
low_reader = refined_df_budget.nsmallest(8, ['budget'])
low_reader

id  budget  revenue                      original_title  runtime  \
1151   287524       1        0                         Fear Clinic       95   
2618    39964       1      100                        Lost & Found       95   
3065    12689       1        0  Angus, Thongs and Perfect Snogging      100   
3581    59296       1     1378             Love, Wedding, Marriage       90   
8944    14373       2       16                        Death Wish 2       88   
2398    39356       3       43                                 Boy       87   
3765    91551       3        0                Death of a Superhero       97   
10050   20701       3       16  Tales from the Darkside: The Movie       93   

                                    production_companies  release_year  
1151   Dry County Films|Anchor Bay Entertainment|Movi...          2014  
2618            Alcon Entertainment|Dinamo Entertainment          1999  
3065   Paramount Pictures|Nickelodeon Movies|Stella-d...          2008  
3581   120dB Films|Scion Films|Voodoo Production Serv...          2011  
8944                                        Cannon Films          1982  
2398   New Zealand Film Commission|Unison Films|Whenu...          2010  
3765      Bavaria Pictures|Grand Pictures|Picture Circle          2011  
10050  Paramount Pictures|Laurel Productions|Darkside...          1990

In [19]:
#Assumption: Since, 3 movies are tied for the 3rd least budget that is why there are 3 answers.
third_lowest = low_reader.drop(low_reader.index[0:5])
third_lowest

id  budget  revenue                      original_title  runtime  \
2398   39356       3       43                                 Boy       87   
3765   91551       3        0                Death of a Superhero       97   
10050  20701       3       16  Tales from the Darkside: The Movie       93   

                                    production_companies  release_year  
2398   New Zealand Film Commission|Unison Films|Whenu...          2010  
3765      Bavaria Pictures|Grand Pictures|Picture Circle          2011  
10050  Paramount Pictures|Laurel Productions|Darkside...          1990

In [20]:
 print("Movie Which Has Third Lowest Budget : ",third_lowest['original_title'])

Movie Which Has Third Lowest Budget :  2398                                    Boy
3765                   Death of a Superhero
10050    Tales from the Darkside: The Movie
Name: original_title, dtype: object


In [21]:
high_reader = refined_df_budget.nlargest(3, ['budget'])
high_reader

id     budget     revenue  \
2244  46528  425000000    11087569   
3375   1865  380000000  1021683000   
7387    285  300000000   961000000   

                                   original_title  runtime  \
2244                            The Warrior's Way      100   
3375  Pirates of the Caribbean: On Stranger Tides      136   
7387     Pirates of the Caribbean: At World's End      169   

                                   production_companies  release_year  
2244                           Boram Entertainment Inc.          2010  
3375  Walt Disney Pictures|Jerry Bruckheimer Films|M...          2011  
7387  Walt Disney Pictures|Jerry Bruckheimer Films|S...          2007

In [22]:
third_highest = high_reader.drop(high_reader.index[0:2])
third_highest

id     budget    revenue                            original_title  \
7387  285  300000000  961000000  Pirates of the Caribbean: At World's End   

      runtime                               production_companies  release_year  
7387      169  Walt Disney Pictures|Jerry Bruckheimer Films|S...          2007

In [23]:
 print("Movie Which Has Third Highest Budget : ",third_highest['original_title'])

Movie Which Has Third Highest Budget :  7387    Pirates of the Caribbean: At World's End
Name: original_title, dtype: object


# Conclusions
### Movie Which Has Highest revenue :  Avatar
### Movie Which Has Lowest revenue  :  Shattered Glass
### Average runtime of movies in the year 2006 : 101.195467
### Any 3 production companies which have invested money in worse revenue movies:
#### 1) Lions Gate Films
#### 2) Gramercy Pictures
#### 3) Universal Pictures
### Movies with the third lowest budget : Boy, Death of a Superhero, Tales from the Darkside: The Movie	(TIED)
### Movie with the third highest budget : Pirates of the Caribbean: At World's End